ui/src/track-layout/track-layout-model.tsx

```diff
@@ -27,6 +27,7 @@ import { exhaustiveMatchingGuard } from 'utils/type-utils';
 import { Brand } from 'common/brand';
 
 export type LayoutState = 'IN_USE' | 'NOT_IN_USE' | 'PLANNED' | 'DELETED';
+export type LocationTrackLayoutState = 'BUILT' | LayoutState;
```

Comment:
Frontissa kaikki se mikä bäkkärillä aiheutti tuskaa onkin miljoona kertaa helpompaa

Reply:
Toisaalta tässä on bugilaisen mahdollisuus: Jos LayoutStatea muutetaan bäkkärillä tavalla, että se sisältää myös variantin, jota ei taas haluttaisi LocationTrackkiin, niin nyt se päätyykin frontissa myös LocationTrackLayoutStateen, vaikka frontissa muuttaisi vain LayoutStatea. Toki riski tähän on melko minimaalinen, kun ovat esim. näin lähekkäin määriteltyjä, muttei sitä silti välttämättä flow-tilassa huomaisi.

Itse suosisin frontin standardoimista bäkkäriä vastaavaksi, eli kopioisin arvolistan, kun ovat kerran tulkittu erilaisiksi asioiksi.

Reply:
`LocationTrackLayoutState` on konseptuaalisesti ihan tarkoituksella `LayoutState`:n superset, ja se on mielestäni tarkoitettu sellaiseksi. Jos ne haluttaisiin erottaa toisistaan siten että pohja-`LayoutState` voi sisältää jotain tiloja joita erikoistukset eivät voi sisältää, niin mielestäni silloin pitää miettiä paljon enemmänkin juttuja uudestaan (mm. tyyppien nimeämiset, koska silloin mielestäni molemmat eivät saisi olla enää `LayoutState`:ja.) Pikemminkin olisin halunnut tehdä tämän ihan kautta linjan niin, että `LocationTrackLayoutState` olisi peritty `LayoutState`:sta koska mun mielestäni näillä on niin syvä yhteys, mutta se ei ole mahdollista koska Kotlinin tai pg:n enumeita ei voi periä.